packages/babel-plugin-transform-regenerator/src/regenerator/emit.ts

```diff
@@ -1,50 +1,24 @@
-/* eslint-disable @typescript-eslint/no-use-before-define */
 /* eslint-disable no-case-declarations */
 import assert from "node:assert";
 import * as leap from "./leap.ts";
 import * as meta from "./meta.ts";
 import * as util from "./util.ts";
 
-const hasOwn = Object.prototype.hasOwnProperty;
+import type { NodePath, Visitor } from "@babel/core";
+import { types as t } from "@babel/core";
 
-export function Emitter(this: any, contextId: any) {
-  assert.ok(this instanceof Emitter);
-
-  util.getTypes().assertIdentifier(contextId);
-
-  // Used to generate unique temporary names.
-  (this as any).nextTempId = 0;
-
-  // In order to make sure the context object does not collide with
-  // anything in the local scope, we might have to rename it, so we
-  // refer to it symbolically instead of just assuming that it will be
-  // called "context".
-  (this as any).contextId = contextId;
-
-  // An append-only list of Statements that grows each time this.emit is
-  // called.
-  (this as any).listing = [];
-
-  // A sparse array whose keys correspond to locations in this.listing
-  // that have been marked as branch/jump targets.
-  (this as any).marked = [true];
-
-  (this as any).insertedLocs = new Set();
-
-  // The last location will be marked when this.getDispatchLoop is
-  // called.
-  (this as any).finalLoc = (this as any).loc();
-
-  // A list of all leap.TryEntry statements emitted.
-  (this as any).tryEntries = [];
-
-  // Each time we evaluate the body of a loop, we tell this.leapManager
-  // to enter a nested loop context that determines the meaning of break
-  // and continue statements therein.
-  (this as any).leapManager = new (leap.LeapManager as any)(this);
-}
-
-const Ep = Emitter.prototype;
+type Completion =
+  | {
+      type: "normal";
```

Comment:
It seems like normal is never used. Let's remove it and call the type "AbruptCompletion" (https://tc39.es/ecma262/#sec-completion-record-specification-type).